web/src/app/dialogs/new-inspection/new-inspection.component.ts

```diff
@@ -249,26 +239,92 @@ export class NewInspectionDialogComponent implements OnDestroy {
 
   public selectedStepChange(stepIndex: number) {
     if (stepIndex === NewInspectionDialogComponent.STEP_INDEX_PARAMETER_INPUT) {
-      this.dryrunRequest.next(this.currentValues.value);
+      this.dryrunRequest.next({});
     }
   }
 
   public toggleFeature(featureId: string) {
     this.featureToggleRequest.next(featureId);
   }
 
+  public run() {
+    this.hadRun.set(true);
+    of(void 0)
+      .pipe(
+        takeUntil(this.destoroyed),
+        withLatestFrom(this.currentTaskClient, this.store.watchAll()),
+        take(1),
+        switchMap(([, client, values]) => client.run(values)),
+      )
+      .subscribe(() => {
+        this.extension.notifyLifecycleOnInspectionStart();
+        this.dialogRef.close({
+          inspectionTaskStarted: true,
+        });
+      });
+  }
+
   /**
-   * track function for ngFor loop in displaying fields.
-   * Need this track mechanism not to recreate form fields and lose focus due to the field collection regeneration.
+   * Convert the array of form fields to the flatten map of default values.
    */
-  public fieldCollectionTrack(
-    index: number,
-    field: InspectionMetadataFormField,
-  ): string {
-    return field.id;
+  private flattenDefaultValues(parameters: ParameterFormField[]): {
+    [key: string]: unknown;
+  } {
+    let result: { [key: string]: unknown } = {};
+    for (const parameter of parameters) {
+      switch (parameter.type) {
+        case ParameterInputType.Text:
+          result[parameter.id] = parameter.default;
+          break;
+        case ParameterInputType.Group:
+          result = {
+            ...result,
+            ...this.flattenDefaultValues(parameter.children),
+          };
+          break;
+        default:
+          break;
+      }
+    }
+    return result;
   }
 
-  public run() {
-    this.runRequest.next(null);
+  /**
+   * Count the count of fields with error.
```

Comment:
nit: maybe simply say "Count error fields" or "Count fields that contain errors" or something like that?

Reply:
Fixed